### DISCUSSION TITLE
Migration Rollback Preview Option

### DISCUSSION THREAD

**Discussion by myckhel** [laravel/framework#33657] [General]
Hello,

I would like to request a feature for the `migrate:rollback` command.
I would like to parse an optional argument to the `migrate:rollback --preview` command which will list the migrations that the rollback command will affect.

For Example:

`php artisan migrate:rollback --preview`

```
**Rollback Migration List**

2013_05_21_140708_create_countries_table

2013_05_21_140748_create_states_table

2014_10_12_000000_create_users_table

2016_06_01_000001_create_oauth_auth_codes_table

2016_06_01_000002_create_oauth_access_tokens_table

2016_06_01_000003_create_oauth_refresh_tokens_table
```

Thanks For The Consideration.

**Comment by ahinkle** [comment]
`--prevent` already does this by displaying the SQL queries that would be run. What are you suggesting that would provide a benefit over `prevent`?

**Reply by myckhel** [reply]
I tried to run the command with the `php artisan migrate:rollback --prevent` option to see it in action but it failed with error
`The "--prevent" option does not exist.`

**Reply by myckhel** [reply]
i Suggested That i see the list of last batch of migrations  file name.

For Example 

if i have in the database table migrations
```
ID,                Migration,                            Batch

"1","2013_05_21_140708_create_countries_table","1"
"2","2013_05_21_140748_create_states_table","1"
"3","2014_10_12_000000_create_users_table","1"
"4","2016_06_01_000001_create_oauth_auth_codes_table","1"
"5","2016_06_01_000002_create_oauth_access_tokens_table","1"
"6","2016_06_01_000003_create_oauth_refresh_tokens_table","1"
"7","2016_06_01_000004_create_oauth_clients_table","1"
"8","2016_06_01_000005_create_oauth_personal_access_clients_table","1"
"9","2019_08_19_000000_create_failed_jobs_table","1"
"10","2019_08_20_151225_create_messages_table","1"
"11","2019_10_11_114900_create_jobs_table","1"
"12","2020_03_10_120541_create_hides_table","1"
"13","2020_03_17_141613_create_admins_table","1"
"14","2020_05_21_133532_create_media_table","1"
"15","2020_05_21_181727_create_metas_table","1"
"16","2020_05_21_182609_create_legals_table","1"
"17","2020_05_21_191115_create_contacts_table","1"
"18","2020_05_25_125602_create_reports_table","1"
"19","2020_05_25_130315_create_notifications_table","1"
"20","2020_05_25_130654_create_user_requests_table","1"
"21","2020_05_25_131114_create_user_notifications_table","1"
"22","2020_05_25_134521_create_likes_table","1"
"23","2020_02_26_173203_create_websockets_statistics_entries_table","2"
"25","2020_07_06_135604_create_permission_tables","3"
"26","2020_07_13_094906_create_user_subscriptions_table","4"
"28","2020_07_24_112645_create_check_mobi_verifications_table","5"
```

if I run `php artisan migrate:rollback --preview`

the output should give me something similar to below

```
Migrations To be affected:
2020_07_24_112645_create_check_mobi_verifications_table
```

**OR**

if I run `php artisan migrate:rollback --confirm`

the output should give me a confirmation prompt something similar to below

```
Migrations To be affected:
2020_07_24_112645_create_check_mobi_verifications_table

Do You want to proceed (y/n)? :
```

Why should it give me above out put?
Because the migration is in the last batch which is batch 5.


Why this feature?
Because sometimes developers would like to see which migrations will be affected when they run the `migrate:rollback` command

so that they dont accidentally rollback migrations they were not intended to rollback.

Thanks for your response in advance.
